include/cynq/accelerator.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * See LICENSE for more information about licensing
+ *
+ * Copyright 2023
+ * Author: Luis G. Leon-Vega <luis.leon@ieee.org>
+ *         Diego Arturo Avila Torres <diego.avila@uned.cr>
+ *
+ */
+#pragma once
+#include "cynq/enums.hpp"
+#include "cynq/status.hpp"
+
+namespace cynq {
+class IAccelerator {
+ public:
+  virtual ~IAccelerator() {}
+  /**
+   * @brief Start method
+   * This starts the accelerator.
+   *
+   * @param mode One of the values in the StartMode enum class
+   * present in the enums.hpp file.
+   *
+   * @return Status
+   */
+  virtual Status Start(const StartMode mode) = 0;
+  /**
+   * @brief Stop method
+   * This stops the accelerator.
+   *
+   * @return Status
+   */
+  virtual Status Stop() = 0;
+  /**
+   * @brief GetStatus method
+   * Returns the DeviceStatus one of the values from the
+   * DeviceStatus enum class present in the enums.hpp file.
```

Comment:
This returns the accelerator state by using the DeviceStatus. This reads the control register flags.